### PR TITLE
Add implementation text for zones.py

### PR DESCRIPTION
**Pull request by mikepjohnson**
## What is the change?
Design and implementation detail added for zones.py.

## Why is the change being made?
This is part of the ongoing SQA effort.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/zones.py [hunk 1]
```diff
@@ -32,6 +32,18 @@ class Zone:
     .. impl:: A user can define a collection of armi locations.
         :id: I_ARMI_ZONE
         :implements: R_ARMI_ZONE
+
+        The Zone class facilitates the creation of a Zone object representing a
+        collection of locations in the Core. A Zone contains a group of locations
+        in the Core, used to subdivide it for analysis. Each location represents
+        an Assembly or a Block, where a single Zone must contain items of the same
+        type (i.e., Assembly or Block). Class methods are provided to add or remove
```
Comment (mgjarrett) [inline armi/reactor/zones.py:40]:
```suggestion
        type (i.e., Assembly or Block). Methods are provided to add or remove
```

`addLoc`, `removeLoc`, etc. are regular instance methods, not class methods.

File: armi/reactor/zones.py [hunk 2]
```diff
@@ -32,6 +32,18 @@ class Zone:
...
+        one or more locations to/from the Zone, and similarly, add or remove one or
+        more items with a Core location (i.e., Assemblies or Blocks) to/from the
+        Zone. In addition, several Class methods are provided to facilitate the
```
Comment (mgjarrett) [inline armi/reactor/zones.py:43]:
```suggestion
        Zone. In addition, several methods are provided to facilitate the
```

File: armi/reactor/zones.py [hunk 3]
```diff
@@ -205,6 +217,16 @@ class Zones:
     .. impl:: A user can define a collection of armi zones.
         :id: I_ARMI_ZONES
         :implements: R_ARMI_ZONES
+
+        The Zones class facilitates the creation of a Zones object representing a
+        collection of Zone objects. Class methods are provided to add or remove one
+        or more Zone to/from the Zones object. Likewise, Class methods are provided
```
Comment (mgjarrett) [inline armi/reactor/zones.py:223]:
```suggestion
        or more Zone to/from the Zones object. Likewise, methods are provided
```

File: armi/reactor/zones.py [hunk 4]
```diff
@@ -205,6 +217,16 @@ class Zones:
     .. impl:: A user can define a collection of armi zones.
         :id: I_ARMI_ZONES
         :implements: R_ARMI_ZONES
+
+        The Zones class facilitates the creation of a Zones object representing a
+        collection of Zone objects. Class methods are provided to add or remove one
```
Comment (mgjarrett) [inline armi/reactor/zones.py:222]:
```suggestion
        collection of Zone objects. Methods are provided to add or remove one
```

File: armi/reactor/zones.py [hunk 5]
```diff
@@ -205,6 +217,16 @@ class Zones:
...
+        or more Zone to/from the Zones object. Likewise, Class methods are provided
+        to validate that the zones are mutually exclusive, obtain the location
+        labels of zones, return the Zone object where a particular Assembly or Block
+        resides, sort the Zone objects alphabetically, and summarize the zone
+        definitions. In addition, Class methods are provided to facilitate the
```
Comment (mgjarrett) [inline armi/reactor/zones.py:227]:
```suggestion
        definitions. In addition, methods are provided to facilitate the
```

File: armi/reactor/zones.py (hunk 1 above)
Comment (john-science) [reply armi/reactor/zones.py:40]: This isn't my PR, but in Python "class method" is the same as "method".

In other languages, "function" and "method" might mean different things.  But in Python a "method" is a function that is part of a class.  And that's just how Python defines those two words.

File: armi/reactor/zones.py (hunk 2 above)
Comment (john-science) [reply armi/reactor/zones.py:43]: This isn't my PR, but in Python "class method" is the same as "method".

In other languages, "function" and "method" might mean different things.  But in Python a "method" is a function that is part of a class.  And that's just how Python defines those two words.

File: armi/reactor/zones.py (hunk 1 above)
Comment (mikepjohnson) [reply armi/reactor/zones.py:40]: Thanks Mike and John, I'm good with the suggested changes and will update all instances in the PR of "class method" to "method" for consistency.

File: armi/reactor/zones.py (hunk 3 above)
Comment (mikepjohnson) [reply armi/reactor/zones.py:223]: Changes accepted and committed.

File: armi/reactor/zones.py (hunk 4 above)
Comment (mikepjohnson) [reply armi/reactor/zones.py:222]: Changes accepted and committed.

**Review by mgjarrett — APPROVED** [review]

**Pull request merged**